packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/studio/services/DefaultDomainResourceProvider.java

```diff
@@ -37,6 +41,12 @@ public class DefaultDomainResourceProvider implements IDefaultDomainResourceProv
 
     private static final String DOMAIN_DOCUMENT_NAME = "DomainNewModel";
 
+    private final List<IMigrationParticipant> migrationParticipants;
```

Comment:
It will not be viable in the long run to ask everybody to get injected with this every time they want to create a new resource.

Reply:
Would setting the correct migrationData in `StudioProjectTemplateInitializer` more acceptable then ?